operator/internal/manifests/storage/schema_test.go

```diff
@@ -0,0 +1,613 @@
+package storage
+
+import (
+	"testing"
+	"time"
+
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BuildSchemaConfigList_NoSchemas(t *testing.T) {
+	specs := []lokiv1beta1.ObjectStorageSchemaSpec{}
+	statuses := []lokiv1beta1.StorageSchemaStatus{}
+
+	_, err := BuildSchemaConfigList(time.Now().UTC(), specs, statuses)
+
+	require.Error(t, err)
+}
+
+func BuildSchemaConfigList_AddSchema_NoStatuses(t *testing.T) {
+	specs := []lokiv1beta1.ObjectStorageSchemaSpec{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+	}
+	statuses := []lokiv1beta1.StorageSchemaStatus{}
+
+	actual, err := BuildSchemaConfigList(time.Now().UTC(), specs, statuses)
+	expected := []lokiv1beta1.ObjectStorageSchemaSpec{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+	}
+
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+func BuildSchemaConfigList_AddSchema_WithPriorSchemas_OutsideUpdateWindow(t *testing.T) {
+	utcTime := time.Date(2021, 9, 1, 0, 0, 0, 0, time.UTC)
+	specs := []lokiv1beta1.ObjectStorageSchemaSpec{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV12,
+			EffectiveDate: "2021-10-01",
+		},
+	}
+	statuses := []lokiv1beta1.StorageSchemaStatus{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+	}
+
+	actual, err := BuildSchemaConfigList(utcTime, specs, statuses)
+	expected := []lokiv1beta1.ObjectStorageSchemaSpec{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV12,
+			EffectiveDate: "2021-10-01",
+		},
+	}
+
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+func BuildSchemaConfigList_AddSchema_WithPriorSchemas_InsideUpdateWindow(t *testing.T) {
```

Comment:
As said above, I suggest to rename this to switch to something `Valid/Invalid` or `Matching/Conflicting`.